app/controllers/posts_controller.rb

```diff
@@ -3,10 +3,26 @@ class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
     @posts = Post.where(author_id: params[:user_id]).order(id: :asc)
-    @posts = @posts.paginate(page: params[:page], per_page: 2)
+    @posts = @posts.paginate(page: params[:page], per_page: 10)
   end
 
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(params.require(:post).permit(:title, :text))
+    @post.author = current_user
+    if @post.save
+      flash[:success] = 'Post created successfully!'
+      redirect_to user_posts_url
+    else
+      flash.now[:error] = 'Error: Post could not be created!'
+      render :new, locals: { post: @post }
+    end
+  end
 end
```

Comment:
- [ ] (Optional) Consider adding the  strong parameters, here is a link to a resource: [Link](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides).

Reply:
Thank you @Munsa1 !
Here I am using strong parameters on line 18, but I haven't encapsulated them. I'll do that definitely when I create #update method to keep my code DRY.